index/scorch/persister.go

```diff
@@ -765,12 +766,15 @@ func (s *Scorch) removeOldData() {
 		s.fireAsyncError(fmt.Errorf("got err removing old bolt snapshots: %v", err))
 	}
 
-	if removed > 0 {
+	// it is safer to attempt the cleaning of stale files even if there were no
+	// snapShot purges.
+	if removed >= 0 {
 		err = s.removeOldZapFiles()
 		if err != nil {
 			s.fireAsyncError(fmt.Errorf("got err removing old zap files: %v", err))
 		}
 	}
+	atomic.AddUint64(&s.stats.TotRemoveOldDataCompleted, 1)
```

Comment:
maybe stats for # of actual files that were removed versus skipped due to ineligibility would give a better picture?  so, some stat that's incremented inside removeOldZapFiles()?

now that I see the removed >= 0 check, wondering if the check is needed at all?  it's always true?

Reply:
I suspect another wrinkle here with the file clean ups. 
ie whenever we retain a couple of older snapshots, we aren't really taking that fact into consideration while deleting the older files. 
s.loadZapFileNames() - as of now it returns the current live files at the root, instead it should give files from all snapshots(older+root) kept for rollback purpose and purge the rest of files?